doc/java.md

```diff
@@ -0,0 +1,298 @@
+Outputting Java Bytecode
+========================
+
+We want to convert flattened Avalanche to jbc.
+It looks like Jacob Stanley has the majority of a decent library for doing this: [https://github.com/jystic/tonic].
+
+There are other libraries which may be better baked, for example hs-java, but I just dislike its imperative way of constructing bytecode.
+I think I will start with Jacob's library, and if I run into too many shortcomings, switch to hs-java.
+
+
+Examples
+========
+
+CountAboveTen
+-------------
+
+Let's start with a very simple one.
+Given the input query:
+```
+feature salary ~> filter value > 10 ~> count
+```
+
+We get the following flattened Avalanche.
+```
+gen$date = DATE
+{
+  init acc$conv$5 = 0 : Int (Resumable);
+  for_facts (gen$fact : Int) in new {
+    let anf$3 = fst# [Int] [DateTime] gen$fact;
+    if (gt# [Int] anf$3 (10 : Int)) {
+      read acc$conv$5 = acc$conv$5;
+      write acc$conv$5 = add# acc$conv$5 (1 : Int);
+    }
+  }
+  read conv$5 = acc$conv$5;
+  return conv$5;
+}
+```
+
+I'm imagining the Java would be something like this.
+```
+void Compute(IcicleMagic icicle)
+{
+    // gen$date = DATE
+    DateTime gen_date = icicle.SnapshotDate();
```

Comment:
FWIW this took me a few seconds to parse because the convention is that functions start with lower case, so I thought you were creating a new `SnapshotDate`.